File: QytetetJava/src/modeloqytetet/Casilla.java
package modeloqytetet;

/**
 *
 * @author aurelia
 */
public abstract class Casilla {
    private int numeroCasilla = 0;
    private TipoCasilla tipo;
    
    public Casilla(int nC, TipoCasilla t){
        numeroCasilla = nC;
        tipo = t;
    }
     
    public TipoCasilla getTipo(){
        return tipo;
    }
    
    // Lo pongo public para que sea accesible desde el controlador
    public int getNumeroCasilla(){
        return numeroCasilla;
    }
    
    abstract boolean soyEdificable();
    
    @Override
    public String toString(){
        return "Casilla{" + " número de casilla=" + Integer.toString(numeroCasilla) +
                ", tipo de casilla=" + tipo + "}\n";
    }
}

File: QytetetRuby/lib/casilla.rb
#encoding: utf-8

module ModeloQytetet
  class Casilla
    def initialize(nc, t)
      @numeroCasilla = nc
      @tipo = t
    end
    
    attr_reader :numeroCasilla, :tipo
    
    def to_s
      "Casilla: \n número de casilla= #{@numeroCasilla} \n tipo de casilla= #{@tipo} \n"
    end
  end
end
